src/gui/guiChatConsole.cpp

```diff
@@ -773,6 +798,52 @@ bool GUIChatConsole::weblinkClick(s32 col, s32 row)
 	return false;
 }
 
+#ifdef USE_CURL
+// This is a function for CURL to write the page's html into a string
+size_t writeHtml(void* contents, size_t size, size_t nmemb, std::string* strptr) {
+    size_t totalSize = size * nmemb;
+    strptr->append((char*)contents, totalSize);
+    return totalSize;
+}
+#endif
+
+void GUIChatConsole::weblinkClickTitle(s32 col, s32 row){
+	std::ostringstream msg;
+	msg << " * ";
+#ifdef USE_CURL
+	std::string weblink, html, title;
+	weblink = getLinkAt(col, row);
```

Comment:
it is very likely not thread safe to access the console buffer from another thread
just pass `std::string weblink` directly to the thread